spec/virtual_includes_spec.rb

```diff
@@ -0,0 +1,180 @@
+describe ActiveRecord::VirtualAttributes::VirtualIncludes do
+  before do
+    Author.destroy_all
+    Book.destroy_all
+    Author.create_with_books(3).books.first.create_bookmarks(2)
+  end
+
+  # books.map(&:author_name)
+  let(:foos) { %w[foo foo foo] }
+  let(:named_books) { Book.where.not(:name => nil).order(:id).load }
+  # # authors.map(&:named_books) (each of 1 author has an array of named books)
```

Comment:
Also, looking at the implementation of the `preload_values` matcher, specifically:

```ruby
expected_values = Array(expected_values) unless expected_values.kind_of?(Array)
```

Is `named_bookss` even needed, since it would just get mapped to an array anyway?

Reply:
it didn't kick in. It couldn't distinguish between `[value]` and `[[value]]`

thanks for info on this one.